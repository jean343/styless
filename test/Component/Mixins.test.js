import React from 'react';
import renderer from 'react-test-renderer';
import styled, {css} from "styled-components";
import 'jest-styled-components';

test('Support inline mixins', () => {
    const Div = styled.div`
        .my-mixin {
          color: @color;
        }
        .my-other-mixin() {
          background: @background;
        }
        .class {
          .my-mixin();
          .my-other-mixin();
        }
	`;
    expect(renderer.create(<Div color="red" background="white"/>).toJSON()).toMatchSnapshot();
});