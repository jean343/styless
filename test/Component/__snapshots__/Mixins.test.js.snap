// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Support inline mixins 1`] = `
.c0 .my-mixin {
  color: red;
}

.c0 .class {
  color: red;
  background: white;
}

<div
  className="c0"
  color="red"
/>
`;
